python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,502 @@
+                   "description": "find z from z = ax + by + c"
+                   }
+    pc.register_scalar_function(varargs_function,
+                                func_name,
+                                varargs_doc,
+                                {
+                                    "array1": pa.int64(),
+                                    "array2": pa.int64(),
+                                    "array3": pa.int64(),
+                                    "array4": pa.int64(),
+                                    "array5": pa.int64(),
+                                },
+                                pa.int64())
+    return varargs_function, func_name
+
+
+@pytest.fixture(scope="session")
+def random_with_udf_ctx_func_fixture():
+    def random_with_udf_ctx(context, one, two):
+        proxy_pool = pa.proxy_memory_pool(context.memory_pool)
+        ans = pc.add(one, two, memory_pool=proxy_pool)
+        res = pa.array([ans.as_py()], memory_pool=proxy_pool)
+        return res
+    in_types = {"one": pa.int64(),
+                "two": pa.int64(),
+                }
+    func_doc = {
+        "summary": "test udf context",
+        "description": "udf context test"
+    }
+    func_name = "test_udf_context"
+    pc.register_scalar_function(random_with_udf_ctx,
+                                func_name, func_doc,
+                                in_types,
+                                pa.int64())
+    return random_with_udf_ctx, func_name
+
+
+@pytest.fixture(scope="session")
+def output_check_func_fixture():
+    # The objective of this fixture is to evaluate,
+    # how the UDF interface respond to unexpected
+    # output types. The types chosen at the test
+    # end are either of different Arrow data type
+    # or non-Arrow type.
+    def output_check(ctx, array):
+        ar = pc.call_function("add", [array, 1])
+        ar = ar.cast(pa.int32())
+        return ar
+    func_name = "test_output_value"
+    in_types = {"array": pa.int64()}
+    out_type = pa.int64()
+    doc = {
+        "summary": "add function scalar",
+        "description": "add function"
+    }
+    pc.register_scalar_function(output_check, func_name, doc,
+                                in_types, out_type)
+    return output_check, func_name
+
+
+@pytest.fixture(scope="session")
+def nullary_check_func_fixture():
+    # this needs to return array values
+    def nullary_check(context):
+        return pa.array([42] * context.batch_length, type=pa.int64(),
+                        memory_pool=context.memory_pool)
+
+    func_doc = {
+        "summary": "random function",
+        "description": "generates a random value"
+    }
+    func_name = "test_random_func"
+    pc.register_scalar_function(nullary_check,
+                                func_name,
+                                func_doc,
+                                {},
+                                pa.int64())
+
+    return nullary_check, func_name
+
+
+def add_const(ctx, scalar):
```

Comment:
Can this be moved to where it's actually used?

Reply:
Ah yes.